emmet-core/emmet/core/qc_tasks.py

```diff
@@ -0,0 +1,616 @@
+        None, description="the type of QChem calculation"
+    )
+
+    orig_inputs: Optional[Union[CalculationInput, Dict[str, Any]]] = Field(
+        {}, description="Summary of the original Q-Chem inputs"
+    )
+
+    input: Optional[InputDoc] = Field(
+        None,
+        description="The input molecule and calc parameters used to generate the current task document.",
+    )
+
+    output: Optional[OutputDoc] = Field(
+        None,
+        description="The exact set of output parameters used to generate the current task document.",
+    )
+
+    # TODO: Implement entry dict
+
+    custodian: Optional[List[CustodianDoc]] = Field(
+        None,
+        title="Calcs reversed data",
+        description="Detailed custodian data for each QChem calculation contributing to the task document.",
+    )
+
+    critic2: Optional[Dict[str, Any]] = Field(
+        None, description="Outputs from the critic2 calculation if performed"
+    )
+
+    custom_smd: Optional[Union[str, Dict[str, Any]]] = Field(
+        None,
+        description="The seven solvent parameters necessary to define a custom_smd model",
+    )
+
+    additional_fields: Optional[Dict[str, Any]] = Field(
+        None, description="Any miscellaneous fields passed to the pydantic model"
+    )
+
+    # TODO some sort of @validator s if necessary
+
+    @classmethod
+    def from_directory(
+        cls: Type[_T],
+        dir_name: Union[Path, str],
+        store_additional_json: bool = True,
+        additional_fields: Dict[str, Any] = None,
+        **qchem_calculation_kwargs,
+    ) -> _T:
+        """
+        Create a task document from a directory containing QChem files.
+
+        Parameters
+        ----------
+        dir_name
+            The path to the folder containing the calculation outputs.
+        store_additional_json
+            Whether to store additional json files in the calculation directory.
+        additional_fields
+            Dictionary of additional fields to add to output document.
+        **qchem_calculation_kwargs
+            Additional parsing options that will be passed to the
+            :obj:`.Calculation.from_qchem_files` function.
+
+        Returns
+        -------
+        QChemTaskDoc
+            A task document for the calculation
+        """
+        logger.info(f"Getting task doc in: {dir_name}")
+
+        additional_fields = {} if additional_fields is None else additional_fields
+        dir_name = Path(dir_name)
+        task_files = _find_qchem_files(dir_name)
+
+        if len(task_files) == 0:
+            raise FileNotFoundError("No QChem files found!")
+
+        critic2 = {}
+        custom_smd = {}
+        calcs_reversed = []
+        for task_name, files in task_files.items():
+            if task_name == "orig":
+                continue
+            else:
+                calc_doc = Calculation.from_qchem_files(
+                    dir_name, task_name, **files, **qchem_calculation_kwargs
```

Comment:
The `**files` construction might be brittle, because you are depending on 1) there only being two keys in each task and 2) them being ordered correctly. I suggest explicitly setting the input and output file kwargs according to the respective keys